File: pset6/credit/credit.py
from cs50 import get_string

card = get_string("Number: ")
i = 0
sum1 = 0
sum2 = 0

for c in card[::-1]:

    if i % 2 == 0:
        sum1 += int(c)
    else:
        if int(c) * 2 < 10:
            sum2 += int(c) * 2;
        else:
            sum2+= (int(c) * 2) % 10;
            sum2+= (int(c) * 2) // 10;
    i += 1

if (sum1 + sum2) % 10 == 0:
    valid = True
else:
    valid = False

if valid:
    if card[0] == '4' and (len(card) == 16 or 13):
        print("VISA\n")
    elif card[0] == '5' and (card[1] == '1' or '2' or '3' or '4'or '5') and len(card) == 16:
        print("MASTERCARD\n")
    elif card[0] == '3' and (card[1] == '4' or '7') and len(card) == 15:
        print("AMEX\n")
    else:
        print("INVALID\n")
elif valid == False:
    print("INVALID\n")

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void){

    int aux = 0;
    long sum = 0;
    long sumprod = 0;
    int digits = 0;
    int firstnumber = 0;

    long num = get_long("Insert your credit card number:\n");

    while(num!=0)
    {
        digits++;
        if(aux==0)
        {
            sum += num % 10;
            aux++;
        }
        else
        {
            if(2*(num%10)<10)
            {
            sumprod += 2*(num % 10);
            }
            else
            {
            sumprod+=(2*(num % 10))/10;
            sumprod+=(2*(num % 10))%10;
            }
            aux--;
        }
        num = num/10;

        if(num < 100 && num >10)
        {
            firstnumber = num;
        }
    }


    if((sum+sumprod)%10==0)
    {

        if(firstnumber/10==4 && (digits == 13 || digits==16))
        {
            printf("VISA\n");
        }
        else if ((firstnumber == 51 || firstnumber == 52||firstnumber == 53||firstnumber == 54 || firstnumber == 55 ) && digits == 16)
        {
            printf("MASTERCARD\n");
        }
        else if ((firstnumber == 34 ||firstnumber == 37) && digits == 15)
        {
            printf("AMEX\n");
        }
        else
        {
            printf("INVALID\n");
        }


    }
    else
    {
        printf("INVALID\n");
    }


    /*printf("sum: %li   sumprod %li digits %i first: %i" , sum, sumprod, digits, firstnumber);*/


}
